library/Faktory/Worker.hs

```diff
@@ -191,10 +191,11 @@ processorLoop f = do
     Right Nothing -> liftIO $ threadDelaySeconds $ settingsIdleDelay workerSettings
     Right (Just job) ->
       processAndAck job
-        `catches` [ Handler $ \(ex :: WorkerHalt) -> throw ex
-                  , Handler $ \(ex :: SomeException) ->
-                    failJob job $ T.pack $ show ex
-                  ]
+      `withException` (\(ex :: SomeException) ->
+        case fromException ex of
+          Just (e :: WorkerHalt) -> throw e
```

Comment:
I don't think you have to throw because `withException` will propagate it automatically.  However, I think this just interrupts that so this is functionally the same thing.